### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by alexsb**
Mainly added documentation to vector and table.

### CODE REVIEW

File: src/data.ts [hunk 1]
```diff
@@ -5,6 +5,11 @@
  **************************************************************************** */
 /**
  * Created by Samuel Gratzl on 04.08.2014.
+ *
+ * This file provides access functions for dataset stored on a server, including functions to list all datasets,
```
Comment (sgratzl) [inline src/data.ts]: file -> module

File: src/table/ITable.ts [hunk 2]
```diff
@@ -27,12 +27,22 @@ export interface ITableColumn<D extends IValueTypeDesc> {
...
+ */
 export interface ITableDataDescription extends IDataDescription {
   readonly idtype: string;
   readonly size: number[];
   readonly columns: IAnyTableColumn[];
 }
 
+/**
+ * A table is a data structure made up of rows and columns. In a table, the elements within a column are always
+ * of the same data type; the data types between different columns can vary. For example, the first row in a
```
Comment (sgratzl) [inline src/table/ITable.ts]: "the first row" -> column

File: src/table/ITable.ts [hunk 3]
```diff
@@ -96,6 +106,11 @@ export interface ITable extends IDataType {
    */
   data(range?: RangeLike): Promise<IValueType[][]>;
 
+  /**
+   * Returns the data of the named column as an array of an array with data
```
Comment (sgratzl) [inline src/table/ITable.ts]: actually it should return a `T[]` - maybe a bug

Comment (sgratzl) [reply src/table/ITable.ts]: was a server bug, is fixed

**Comment by alexsb** [comment]
Can we merge that at some point? I'm happy to get comments on the documentation, but I believe that what's there is already more helpful than what was there before.

**Comment by mstreit** [comment]
Yes, this is more helpful compared to what we currently have and it's documentation only. So I guess it's fine to merge it now so that others in Alex's team can profit from this.

**Pull request merged**